rosbag2_compression/test/rosbag2_compression/test_sequential_compression_reader.cpp

```diff
@@ -0,0 +1,103 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gmock/gmock.h>
+
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "rosbag2_compression/sequential_compression_reader.hpp"
+#include "rosbag2_cpp/reader.hpp"
+
+#include "../../rosbag2_cpp/test/rosbag2_cpp/mock_converter_factory.hpp"
+#include "../../rosbag2_cpp/test/rosbag2_cpp/mock_metadata_io.hpp"
+#include "../../rosbag2_cpp/test/rosbag2_cpp/mock_storage.hpp"
+#include "../../rosbag2_cpp/test/rosbag2_cpp/mock_storage_factory.hpp"
+
+using namespace testing;  // NOLINT
+
+class SequentialCompressionReaderTest : public Test
+{
+public:
+  SequentialCompressionReaderTest()
+  : storage_factory_(std::make_unique<StrictMock<MockStorageFactory>>()),
+    storage_(std::make_shared<NiceMock<MockStorage>>()),
+    converter_factory_(std::make_shared<StrictMock<MockConverterFactory>>()),
+    metadata_io_(std::make_unique<NiceMock<MockMetadataIo>>()),
+    storage_serialization_format_("rmw1_format")
+  {
+    topic_with_type_ = rosbag2_storage::TopicMetadata{
+      "topic", "test_msgs/BasicTypes", storage_serialization_format_};
+    auto topics_and_types = std::vector<rosbag2_storage::TopicMetadata>{topic_with_type_};
+    auto message = std::make_shared<rosbag2_storage::SerializedBagMessage>();
+    message->topic_name = topic_with_type_.name;
+
+    EXPECT_CALL(*storage_, get_all_topics_and_types())
+    .Times(AtMost(1)).WillRepeatedly(Return(topics_and_types));
+    EXPECT_CALL(*storage_, read_next()).WillRepeatedly(Return(message));
```

Comment:
I think you actually want `ON_CALL`.
iirc: `EXPECT_CALL` is more useful if you are checking if the function is being called